packages/core/src/typeschema/validation.ts

```diff
@@ -247,6 +271,39 @@ class ResourceValidator {
     }
   }
 
+  private constraintsCheck(value: TypedValue, element: ElementValidator, path: string): void {
+    const constraints = element.constraints;
+    for (const constraint of constraints) {
+      if (constraint.severity !== 'error' || constraint.key in skippedConstraintKeys) {
+        continue;
+      } else {
+        const expression = this.isExpressionTrue(constraint, value, path);
+        if (!expression) {
+          this.issues.push(createStructureIssue(path, `Constraint ${constraint.key} failed`));
+          return;
+        }
+      }
+    }
+  }
+
+  private isExpressionTrue(constraint: Constraint, value: TypedValue, path: string): boolean {
+    try {
+      const evalValues = evalFhirPathTyped(constraint.expression, [value], {
+        context: value,
+        resource: toTypedValue(this.currentResource[this.currentResource.length - 1]),
+        rootResource: toTypedValue(this.rootResource),
+        ucum: toTypedValue('http://unitsofmeasure.org'),
+      });
+      if (constraint.key === 'opd-3' && evalValues[0].value === false) {
+        console.log(constraint.expression, value);
+      }
+      return evalValues.every((evalValue) => evalValue.value === true);
+    } catch (e: any) {
+      this.issues.push(createStructureIssue(path, `Constraint ${constraint.key} failed with error: ${e.message}`));
```

Comment:
I wouldn't call this a structure issue: since it's an exception in the FHIRPath evaluation, it might be better classed as an internal server error